packages/utils/src/feature_toggle.rs

```diff
@@ -0,0 +1,393 @@
+        Ok(())
+    }
+
+    fn unpause<S: Storage, T: Serialize>(storage: &mut S, features: Vec<T>) -> StdResult<()> {
+        for f in features {
+            Self::set_feature_status(storage, f, Status::NotPaused)?;
+        }
+
+        Ok(())
+    }
+
+    fn get_pauser<S: ReadonlyStorage>(storage: &S, key: HumanAddr) -> StdResult<Option<u8>> {
+        let feature_store =
+            ReadonlyBucket::multilevel(&[Self::STORAGE_KEY, PREFIX_PAUSERS], storage);
+        feature_store.may_load(key.0.as_bytes())
+    }
+
+    fn set_pauser<S: Storage>(storage: &mut S, key: HumanAddr) -> StdResult<()> {
+        let mut feature_store = Bucket::multilevel(&[Self::STORAGE_KEY, PREFIX_PAUSERS], storage);
+        feature_store.save(key.0.as_bytes(), &1_u8 /* value is insignificant */)
+    }
+
+    fn remove_pauser<S: Storage>(storage: &mut S, key: HumanAddr) {
+        let mut feature_store: Bucket<S, u8> =
+            Bucket::multilevel(&[Self::STORAGE_KEY, PREFIX_PAUSERS], storage);
+        feature_store.remove(key.0.as_bytes())
+    }
+
+    fn get_feature_status<S: ReadonlyStorage, T: Serialize>(
+        storage: &S,
+        key: T,
+    ) -> StdResult<Option<Status>> {
+        let feature_store =
+            ReadonlyBucket::multilevel(&[Self::STORAGE_KEY, PREFIX_FEATURES], storage);
+        feature_store.may_load(&cosmwasm_std::to_vec(&key)?)
+    }
+
+    fn set_feature_status<S: Storage, T: Serialize>(
+        storage: &mut S,
+        key: T,
+        item: Status,
+    ) -> StdResult<()> {
+        let mut feature_store = Bucket::multilevel(&[Self::STORAGE_KEY, PREFIX_FEATURES], storage);
+        feature_store.save(&cosmwasm_std::to_vec(&key)?, &item)
+    }
+}
+
+impl FeatureToggle {
+    pub fn require_not_paused<S: Storage, T: Serialize + Display + Clone>(
+        storage: &S,
+        features: Vec<T>,
+    ) -> StdResult<()> {
+        for feature in features {
+            let status = Self::get_feature_status(storage, feature.clone())?;
+            match status {
+                None => {
+                    return Err(StdError::generic_err(format!(
+                        "feature toggle: unknown feature '{}'",
+                        feature
+                    )))
+                }
+                Some(s) => match s {
+                    Status::NotPaused => {}
+                    Status::Paused => {
+                        return Err(StdError::generic_err(format!(
+                            "feature toggle: feature '{}' is paused",
+                            feature
+                        )));
+                    }
+                },
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn handle_pause<S: Storage, A: Api, Q: Querier, T: Serialize>(
+        deps: &mut Extern<S, A, Q>,
+        env: Env,
+        features: Vec<T>,
+    ) -> HandleResult {
+        if Self::get_pauser(&deps.storage, env.message.sender)?.is_none() {
+            return Err(StdError::unauthorized());
+        }
```

Comment:
If you change `get_pauser` to take the address by reference, you can also take `Env` by reference here and in other places

Reply:
I tend to agree 👍🏼